packages/rum-core/src/boot/preStartRum.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  BoundedBuffer,
+  display,
+  type DeflateWorker,
+  canUseEventBridge,
+  displayAlreadyInitializedError,
+  willSyntheticsInjectRum,
+  noop,
+  timeStampNow,
+  clocksNow,
+  assign,
+} from '@datadog/browser-core'
+import {
+  validateAndBuildRumConfiguration,
+  type RumConfiguration,
+  type RumInitConfiguration,
+} from '../domain/configuration'
+import type { CommonContext } from '../domain/contexts/commonContext'
+import type { ViewOptions } from '../domain/view/trackViews'
+import type { RumPublicApiOptions, Strategy } from './rumPublicApi'
+import type { StartRumResult } from './startRum'
+
+export function createPreStartStrategy(
+  { ignoreInitIfSyntheticsWillInjectRum, startDeflateWorker }: RumPublicApiOptions,
+  getCommonContext: () => CommonContext,
+  doStartRum: (
+    initConfiguration: RumInitConfiguration,
+    configuration: RumConfiguration,
+    deflateWorker: DeflateWorker | undefined,
+    initialViewOptions?: ViewOptions
+  ) => StartRumResult
+): Strategy {
+  const bufferApiCalls = new BoundedBuffer<StartRumResult>()
+  let firstStartViewCall:
+    | { options: ViewOptions | undefined; callback: (startRumResult: StartRumResult) => void }
+    | undefined
+  let deflateWorker: DeflateWorker | undefined
+
+  let cachedInitConfiguration: RumInitConfiguration | undefined
+  let cachedConfiguration: RumConfiguration | undefined
+
+  function tryStartRum() {
+    if (!cachedInitConfiguration || !cachedConfiguration) {
+      return
+    }
+
+    let initialViewOptions: ViewOptions | undefined
+
+    if (cachedConfiguration.trackViewsManually) {
+      if (!firstStartViewCall) {
+        return
+      }
+      // An initial view is always created when starting RUM.
+      // When tracking views automatically, any startView call before RUM start creates an extra
+      // view.
+      // When tracking views manually, we use the ViewOptions from the first startView call as the
+      // initial view options, and we remove the actual startView call so we don't create
+      bufferApiCalls.remove(firstStartViewCall.callback)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** unfinished sentence?‏
```suggestion
      // When tracking views manually, we use the ViewOptions from the first startView call as the
      // initial view options, and we remove the actual startView call so we don't create an extra view.
      bufferApiCalls.remove(firstStartViewCall.callback)
```